core/src/com/serwylo/beatgame/levels/Level.kt

```diff
@@ -121,6 +137,46 @@ class RemoteLevel(private val world: RemoteWorld, private val data: WorldDTO.Lev
 
 }
 
+
+class CustomWorld(levelsData: List<CustomWorldDTO.CustomLevelDTO>): World {
+
+    private val levels = levelsData.map {
+        CustomLevel(this, it.id, it.label, Gdx.files.absolute(it.mp3Path))
+    }
+
+    override fun getId() = "custom"
+
+    override fun getLabel(strings: I18NBundle) = "Your world"
+
+    override fun getLevels() = levels
+
+    override fun getAttribution() = emptyList<Attribution>()
+
+    companion object {
+
+    }
+}
+
+class CustomLevel(private val world: CustomWorld, private val id: String, private val label: String, private val mp3File: FileHandle): Level {
+    override fun getId() = id
+
+    override fun getMp3File(): FileHandle = mp3File
+
+    override fun getLevelDataFile(): FileHandle {
+        val name = "custom-${getMp3File().lastModified()}"
+        return Gdx.files.local(".cache${File.separator}world${File.separator}$name.json")
```

Comment:
Just last modified, but not the filename. Should probably use level.id here too.

Reply:
I think this is from when you only had one level, and it would get overridden when players wanted to try another song. So filename didn't work then, but modified date did. Now we can get away with the `id` of the custom world I think.